sisl/_blender.py

```diff
@@ -0,0 +1,501 @@
+from scipy.spatial.transform import Rotation
+import numpy as np
+from math import acos
+import os
+from ._array import arrayd
+
+try:
+    import bpy
+    from mathutils import Matrix, Vector
+
+    try:
+        import addon_utils
+        INSIDE_BLENDER = True
+        BLENDER_ASPACKAGE = False
+        
+    except ModuleNotFoundError:
+        BLENDER_ASPACKAGE = True
+        INSIDE_BLENDER = False
+
+    BLENDER_AVAIL = True
+
+except ModuleNotFoundError as BLENDER_IMPORT_EXCEPTION:
+    BLENDER_AVAIL = False
+    BLENDER_ASPACKAGE = False
+    INSIDE_BLENDER = False
+
+__all__ = ['BlenderScene', 'blender']
+
+class BlenderScene:
+
+    _POVs = {
+        '+z': (0,0,1),
+        '-z': (0,0,-1),
+        '+y': (0,1,0),
+        '-y': (0,-1,0),
+        '+x': (1,0,0),
+        '-x': (-1,0,0),
+    }
+
+    def __init__(self, objs = [], filepath = None):
+
+        #Define some attributes that will be helpful so that bpy does not need to be imported elsewhere
+        self.ops = bpy.ops
+        self.context = bpy.context
+        self.data = bpy.data
+        self.atomic_blender = lambda: addon_utils.enable("io_mesh_atomic")
+
+        self.filepath = filepath or os.path.join(os.path.expanduser("~"), "image")
+
+        #Clean the scene of any objects that were there (e.g. the default cube)
+        self.clean()
+
+        self.objs = []
+        self.lights = []
+
+        #If objects were provided on initialization, add them to the scene
+        for obj in objs:
+            self.add(obj)
+
+    def update_bounding_box(self):
+        '''
+        Updates the bounding box of the model so that lights and cameras can be placed correctly.
+        '''
+        self.bounding_box = np.array([
+            np.min([np.min(obj.bound_box, axis=0 ) for obj in self.data.objects], axis = 0),
+            np.max([np.max(obj.bound_box, axis=0 ) for obj in self.data.objects], axis = 0),
+        ]).T
+
+        return self
+
+    def lim(self, min_max = None, axis = None):
+
```

Comment:
add comments, possibly extending this name to `set_limits` or just `limits` (for completeness sake).

Reply:
I have renamed it to `get_limits`